CI: Update code for mypy 1.17

New PRs in CI run against mypy 1.17, while the most recent run on main ran against 1.16. mypy now raises the following warning:
```
modin/logging/logger_decorator.py:117: error: Unused "type: ignore" comment  [unused-ignore]
Found 1 error in 1 file (checked 30 source files)
```

It seems like this came from some improvement to type inference, and we should update this line accordingly.

--- a/modin/logging/logger_decorator.py
+++ b/modin/logging/logger_decorator.py
@@ -114,7 +114,7 @@ def enable_logging(
                         )(attr_value)
 
                     setattr(obj, attr_name, wrapped)
-            return obj  # type: ignore [return-value]
+            return obj
         elif isinstance(obj, classmethod):
             return classmethod(decorator(obj.__func__))  # type: ignore [return-value, arg-type]
         elif isinstance(obj, staticmethod):

